File: Link.Pokedex.Client/ClientApp/src/app/models/dto/pokemon-dto.ts
import { TipoDto } from "./tipo-dto";

export class PokemonDto {
  Nombre: string;
  Altura: number;
  Categoria: string;
  Peso: number;
  Habilidad: string;
  Sexo: string;
  Ps: number;
  Ataque: number;
  Defensa: number;
  AtaqueEspecial: number;
  DefensaEspecial: number;
  Velocidad: number;
  Imagen: string;
  Tipo: Array<TipoDto>;
  Debilidad: Array<TipoDto>;

  constructor() {
    this.Nombre = "";
    this.Altura = 0;
    this.Categoria = "";
    this.Peso = 0;
    this.Habilidad = "";
    this.Sexo = "";
    this.Ps = 0;
    this.Ataque = 0;
    this.Defensa = 0;
    this.AtaqueEspecial = 0;
    this.DefensaEspecial = 0;
    this.Velocidad = 0;
    this.Imagen = "";
    this.Tipo = [];
    this.Debilidad = [];
  }
}

File: Link.Pokedex.Api/Infrastructure/Crosscutting/Dto/PokemonDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Link.Pokedex.Api.Infrastructure.Crosscutting.Dto
{
    public class PokemonDto
    {
        public string Nombre { get; set; }
        public decimal? Altura { get; set; }
        public string Categoria { get; set; }
        public decimal? Peso { get; set; }
        public string Habilidad { get; set; }
        public string Sexo { get; set; }
        public int? Ps { get; set; }
        public int? Ataque { get; set; }
        public int? Defensa { get; set; }
        public int? AtaqueEspecial { get; set; }
        public int? DefensaEspecial { get; set; }
        public int? Velocidad { get; set; }
        public string Imagen { get; set; }

        public List<TipoDto> Tipo { get; set; }
        public List<TipoDto> Debilidad { get; set; }
    }
}
